lib/ipam/ipam_block.go

```diff
@@ -53,6 +53,33 @@ func newBlock(cidr cnet.IPNet) allocationBlock {
 		b.Unallocated[i] = i
 	}
 
+	if rsvdAttr != nil {
+		// Reserve IPs based on host reserved attributes.
+		// For example, with windows OS, the following IP addresses of the block are
+		// reserved. This is done by pre-allocating them during initialization
+		// time only.
+		// IPs : x.0, x.1, x.2 and x.bcastAddr (e.g. x.255 for /24 subnet)
+
+		log.Infof("Block %s reserving IPs", b.CIDR.String())
+		// nil attributes
+		attrs := make(map[string]string)
+		attrs["note"] = rsvdAttr.Note
```

Comment:
We've defined common attributes here, probably worth doing "note" there as well so any external controller can reliably get it: https://github.com/projectcalico/libcalico-go/blob/master/lib/ipam/ipam.go#L42-L48

Reply:
Do we rely on `note` for anything? I thought it is just for description purpose.